lib/nm/source.rb

```diff
@@ -35,8 +36,8 @@ def render(template_name, locals = nil)
 
     private
 
-    def source_file_path(template_name)
-      Dir.glob(self.root.join("#{template_name}*")).first
+    def source_file_path(name)
+      Dir.glob(self.root.join(name.end_with?(@ext) ? name : "#{name}*#{@ext}")).first
```

Comment:
@jcredding Here is where I make the glob expect the the template to end with the ext if one is present.  Notice I default the ext to `nil` above.  This allows me to not have to check whether an ext is present of not.  If no extension is present, `end_with?` is false the the string interpolations means the glob ends up being `name*`.

Does this work for you?